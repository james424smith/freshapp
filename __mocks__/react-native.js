import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
