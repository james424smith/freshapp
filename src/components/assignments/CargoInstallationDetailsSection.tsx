import React, { ReactElement } from "react";
import { View } from "react-native";
import Text from "../StyledText";
import styles from "../../styles/AssignmentScreenStyles";
import CargoInstallationComponent from "./CargoInstallationComponent";
import { CargoItemData, CargoDataTable } from "../../interfaces";
import valueOrDefault from "../../common/valueOrDefault";
import { useTheme } from "@react-navigation/native";
type Props = {
  cargoInstallationTable?: CargoDataTable[];
};

const CargoInstallationDetailsSection = (props: Props) => {
  const { cargoInstallationTable } = props;
  const theme = useTheme();
  return (
    <View
      style={{
        ...styles.cargoDetailsContainerAlignment,
        backgroundColor: theme.colors.primary,
      }}
    >
      <View>
        <View style={styles.row}>
          <Text style={{ ...styles.cargoTitle, color: theme.colors.text }}>
            Cargo Installation
          </Text>
        </View>
        <View style={styles.cargoDataContainer}>
          {(valueOrDefault(cargoInstallationTable, []) as CargoDataTable[]).map(
            (cargoItem: CargoDataTable, i: number) => (
              <View key={i}>
                {cargoItem.map(
                  (
                    cargoLineItem: CargoItemData,
                    item: number
                  ): ReactElement => (
                    <CargoInstallationComponent
                      key={item}
                      textLabel={cargoLineItem.label}
                      textValue={cargoLineItem.value}
                    />
                  )
                )}
                <Text />
              </View>
            )
          )}
        </View>
      </View>
    </View>
  );
};

export default CargoInstallationDetailsSection;
