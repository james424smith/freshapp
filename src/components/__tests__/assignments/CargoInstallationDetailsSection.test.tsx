import React from "react";
import { render, cleanup } from "@testing-library/react-native";
import CargoInstallationDetailsSection from "../../assignments/CargoInstallationDetailsSection";
import { vesselDetails } from "../../../../__mocks__/fake-data/assignmentDetails.json";

describe("CargoInstallationDetailsSection snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render CargoInstallationDetailsSection correctly as per the ui made when the  cargoInstallationTable has values", () => {
    const { toJSON } = render(
      <CargoInstallationDetailsSection
        cargoInstallationTable={vesselDetails.cargoInstallation}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("should render CargoInstallationDetailsSection correctly as per the ui made when cargoInstallationTable has no values ", () => {
    const { toJSON } = render(
      <CargoInstallationDetailsSection cargoInstallationTable={[]} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
