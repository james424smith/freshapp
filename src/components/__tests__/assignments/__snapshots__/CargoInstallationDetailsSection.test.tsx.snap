// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CargoInstallationDetailsSection snapshot test should render CargoInstallationDetailsSection correctly as per the ui made when cargoInstallationTable has no values  1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(0, 122, 255)",
      "paddingTop": 22.5,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "marginTop": 34,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "color": "rgb(28, 28, 30)",
              "flexBasis": "90%",
              "fontSize": 46.5,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Cargo Installation
      </Text>
    </View>
    <View
      style={
        Object {
          "margin": 37.5,
        }
      }
    />
  </View>
</View>
`;

exports[`CargoInstallationDetailsSection snapshot test should render CargoInstallationDetailsSection correctly as per the ui made when the  cargoInstallationTable has values 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(0, 122, 255)",
      "paddingTop": 22.5,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "marginTop": 34,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "color": "rgb(28, 28, 30)",
              "flexBasis": "90%",
              "fontSize": 46.5,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Cargo Installation
      </Text>
    </View>
    <View
      style={
        Object {
          "margin": 37.5,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 24,
                  "paddingRight": 22.5,
                },
              ]
            }
          >
            Number
            :
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "maxWidth": "80%",
                },
              ]
            }
          >
            2
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 24,
                  "paddingRight": 22.5,
                },
              ]
            }
          >
            Unit
            :
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "maxWidth": "80%",
                },
              ]
            }
          >
            TO
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 24,
                  "paddingRight": 22.5,
                },
              ]
            }
          >
            Cargo Installation
            :
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "maxWidth": "80%",
                },
              ]
            }
          >
            CRANES
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 24,
                  "paddingRight": 22.5,
                },
              ]
            }
          >
            Capacity
            :
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 26.5,
                  "fontWeight": "bold",
                  "maxWidth": "80%",
                },
              ]
            }
          >
            250
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
